### PR TITLE
add project spec notebook

### PR DESCRIPTION
**Pull request by aays**

### CODE REVIEW

File: fma_project_spec.ipynb [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+   "source": [
+    "**Project title**: Kick out the jams: Classifying song genres\n",
+    "\n",
+    "**Name:** Lina Tran, Joel Ostblom, Madeleine Bonsma-Fisher, Ahmed Hasan\n",
+    "\n",
+    "**E-mail:** A DataCamp connected e-mail (connected in the sense that you could use it to log in to datacamp.com).\n",
+    "\n",
+    "**GitHub username**: linanmt, joelostblom, mbonsma, aays\n",
+    "\n",
+    "**Link to prior writing**: Please supply a link/reference to some of your own already published educational writing. For example, in the form of a blog post, notebook, article, book or internal case study.\n",
```
Comment (joelostblom) [inline fma_project_spec.ipynb]: We can add a link to our lecture folder here, or maybe selected lectures that one of us has authored. Or just link the workshop material we did last week, that's maybe easier.

File: fma_project_spec.ipynb [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Project title**: Kick out the jams: Classifying song genres\n",
+    "\n",
+    "**Name:** Lina Tran, Joel Ostblom, Madeleine Bonsma-Fisher, Ahmed Hasan\n",
+    "\n",
+    "**E-mail:** A DataCamp connected e-mail (connected in the sense that you could use it to log in to datacamp.com).\n",
```
Comment (joelostblom) [inline fma_project_spec.ipynb]: Add all our emails here, the one I use for datacamp is joel.ostblom@gmail.com. @linamnt and @mbonsma, which are yours?

File: fma_project_spec.ipynb [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    "This will be read by the students on the DataCamp platform **before** deciding to start the project.\n",
+    "\n",
+    "It should mention some of the major prerequisites for completing the project \n",
+    "(for example \"familiarity with `pandas` DataFrames\" or \"know how to use the Naive Bayes method from `scikit learn`\")\n",
+    "\n",
+    "> Using a subset of the dataset comprised of two genres (Hip-Hop and Rock), we will train a classifier to distinguish between the two genres based only on track infromation derived from Echonest (now part of Spotify) data for each track. We will first make use of `pandas` and `seaborn` packages in Python for subsetting the data, aggregating information, and creating plots when exploring the data for obvious trends or factors we should be aware of when doing machine learning. Next, we will use of the `scikit-learn` package to predict whether we can correctly classify a song's genre based on features such as danceability, energy, acousticness, tempo etc. We will go over implementations of common algorithms such as logistic regression, support vector machines, k-nearest neighbours clustering, decision trees and so forth.\n",
```
Comment (joelostblom) [inline fma_project_spec.ipynb]: infromation -> information

Last sentence: I think we will only use PCA, logistic regression and decision trees.

File: fma_project_spec.ipynb [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    "\n",
+    "\n",
+    "#### Datasets used ####\n",
+    "\n",
+    "Short description (and ideally links) to the datasets used in the project. This will be read my me (David) only.\n",
+    "\n",
+    "> An open dataset for music analysis - https://github.com/mdeff/fma. The tracks represented in the dataset are all from the Free Music Archive, a large library of free audio downloads. \n",
+    "\n",
+    "TODO: add links to datasets - these can be links to GitHub uploads\n",
```
Comment (joelostblom) [inline fma_project_spec.ipynb]: Is this TODO for our subsetted data sets? you already have a link to the main repo.

File: fma_project_spec.ipynb [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+    "\n",
+    "#### Assumed student background ####\n",
+    "\n",
+    "What background knowledge you assume the student doing this project will have. The more specific the better. This will be read my me (David) only. Please list things like modules, tools, functions, methods, and statistical concepts and jargon.\n",
+    "\n",
+    "Not so useful: \"The student has a basic familiarity with `pandas`.\"\n",
+    "\n",
+    "More useful: \"The student knows how to read in a csv file using `pandas` and how to compute grouped summary statistics using `groupby()`.\"\n",
+    "\n",
+    "> The student knows how to read in csv and json files using `pandas`, and examine correlations across the columns of a `pandas` DataFrame. The student understands basic plotting (scatter plots, pair plots) with `seaborn`. The student is also familiar with implementing both principal component analysis and supervised learning methods (k-fold crossvalidation) in `scikit-learn`."
```
Comment (joelostblom) [inline fma_project_spec.ipynb]: Suggestion for the last sentence:

The student is also familiar with implementing principal component analysis, supervised learning methods (decision trees and logistic regression), and model accuracy scoring using k-fold crossvalidation and  the classification report in scikit-learn.

@linamnt Do we want to talk about precision and recall and plot the curves in the final version?

File: fma_project_spec.ipynb [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+   "metadata": {},
+   "source": [
+    "# 2. Project narrative intro"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## These recommendations are so on point! How does this playlist know me so well?\n",
```
Comment (joelostblom) [inline fma_project_spec.ipynb:71]: Nice intro blurb!

File: fma_project_spec.ipynb [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "tracks = pd.read_csv('./fma-rock-vs-hiphop.csv')\n",
+    "tracks.info()\n",
+    "track_metrics = pd.read_json('./echonest-clean.json', precise_float=True)\n",
```
Comment (joelostblom) [inline fma_project_spec.ipynb:106]: We should either mention or remove `precise_float`, I don't think they would know to use it here.

**Review by joelostblom — CHANGES_REQUESTED** [review]
Looks great overall! Let me know what you think of the suggestions I made and sorry for not getting to this earlier.

File: fma_project_spec.ipynb (hunk 4 above)
Comment (joelostblom) [reply fma_project_spec.ipynb]: I create a link for our data sets https://www.dropbox.com/sh/bab3yu27thfcv45/AAAHoBux36O54Dyg6UUOBv_8a?dl=0

File: fma_project_spec.ipynb (hunk 5 above)
Comment (linamnt) [reply fma_project_spec.ipynb]: I think this is pretty useful on the grand scheme of ML so I lean towarads yes. If it feels like there's too much it's easily mentionable without plotting/going in depth

File: fma_project_spec.ipynb (hunk 1 above)
Comment (aays) [reply fma_project_spec.ipynb]: I just pulled up various Coders/R Course lessons for this - 

[Lina](https://github.com/UofTCoders/studyGroup/blob/gh-pages/lessons/python/classes/Classes_in_Python.ipynb), [Joel](https://github.com/UofTCoders/rcourse/blob/master/lec02-basic-r.Rmd), [Madeleine](https://github.com/UofTCoders/rcourse/blob/master/lec07-pop-models.Rmd), [Ahmed](https://github.com/UofTCoders/studyGroup/blob/gh-pages/lessons/r/dplyrmagrittr/lesson.Rmd)

File: fma_project_spec.ipynb (hunk 2 above)
Comment (aays) [reply fma_project_spec.ipynb]: All good - nabbed these from our email thread with David!

**Comment by aays** [comment]
@joelostblom I've updated the notebook to reflect all of the above feedback, save for the `precise_float` bit - are we mentioning it or removing it?

**Comment by joelostblom** [comment]
Great, thanks @aays ! Let's remove `precise_float` for now, and decide wether to add it back in for the  final version (ill remove it)

We should be good to hit submit now then? @linamnt Do you want to email this Maddy's to David?

**Review by joelostblom — APPROVED** [review]

**Pull request merged**